### ISSUE TITLE
It is possible to change loss in object detection api?

### ISSUE THREAD

**Issue by zychen2016** [tensorflow/models#7817]
Hi.
i want to change loss of object detection for ones of object detection (such as SSD) ,
Q1 : i want to know where do i modify the loss function for SSD ,
Q2 : is it possible to fine-tune ssd_mobilenet on my dataset with my define loss ? is it good or must be train ssd_mobile from scratch with my loss function ?

https://github.com/tensorflow/models/issues/3395

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by zychen2016** [comment]
What is the top-level directory of the model you are using : object detection api
Have I written custom code: want to write
OS Platform and Distribution:Ubuntu18.04
TensorFlow installed from:pip
TensorFlow version:tf1.15.0
Bazel version:-
CUDA/cuDNN version:10.0 CUDNN 7
GPU model and memory:1080Ti
Exact command to reproduce

**Comment by zychen2016** [comment]
Any one could answer this question?

**Comment by zychen2016** [comment]
@tensorflowbutler

**Comment by zychen2016** [comment]
@gsssrao

**Comment by pkulzc** [comment]
1. add your new loss to [losses.py](https://github.com/tensorflow/models/blob/master/research/object_detection/core/losses.py)
2. add its config field to [losses.proto](https://github.com/tensorflow/models/blob/master/research/object_detection/protos/losses.proto) so it can be parsed by config parser.
3. link the config field you added with the new loss you added, in [loss_builder](https://github.com/tensorflow/models/blob/master/research/object_detection/builders/losses_builder.py).

Finally, use your new loss in your config file.
